### PR TITLE
Performance

### PR DESCRIPTION
**Pull request by ybadr99**
- Prepare db to enhance the performance 
- Add indexing 
Screenshot from 2023-07-19 23-39-12
Screenshot from 2023-07-19 23-39-20
Screenshot from 2023-07-19 23-40-40
Screenshot from 2023-07-19 23-46-04
Screenshot from 2023-07-19 23-47-38

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -118,3 +118,11 @@ VALUES (3, 10, '2020-05-24');
...
 INSERT INTO visits (vets_id, animals_id, date_of_visit)
 VALUES (1, 10, '2021-01-11');
+
+
+
+-- This will add 3.594.280 visits considering you have 10 animals, 4 vets, and it will use around ~87.000 timestamps (~4min approx.)
+INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;
+
+-- This will add 2.500.000 owners with full_name = 'Owner <X>' and email = 'owner_<X>@email.com' (~2min approx.)
+insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000), 'owner_' || generate_series(1,2500000) || '@mail.com';
```
Comment (Gambit142) [inline data.sql]: - [ ] Great work inserting data the way it was recommended. However, I noticed in your screenshots that the execution time before improving performance is significantly lower than **_1 second (1000ms)_**. I would advise you to run this query two or three more times, make the `explain analyze` statement and take screenshots of the execution time that are equal to or more than _**1 second (1000ms)**_. 👍 😎 

#### You can check points _**3 and 4**_ of this [project requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_performance_audit.md) 💻

### Hint: 
- Ensure the queries highlighted below are run before the indexing using the `explain analyze` command: 

  - SELECT COUNT(*) FROM visits where animal_id = 4;

  - SELECT * FROM visits where vet_id = 2;

  - SELECT * FROM owners where email = 'owner_18327@mail.com';

- Then ensure that the execution time for each query is equal to or above **_1 second (1000ms)_** 
- Then take a screenshot and replace the old ones with the new screenshots. 
- Then you apply the indexing and run the `explain analyze` command, take the screenshot and attach the screenshots. 👍 💻

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @ybadr99 @WilhelmK109 👋, 

Good job so far!

## Highlights
- Populated database with data ✔️ 
- Created index on required columns ✔️ 
- Use of the `explain analyze` command ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @ybadr99        👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the SQL queries✔️ 
  
  
# Required Changes  🌵  

- [ ] You've done well, However, kindly check some key point note I pointed out to meet the requirements before moving to the next project 👍 

- [ ] The second query screenshot is close or rather has little effect. Kindly make reference to the last reviewers' hints given to have a variation to show efficiency in the second query.

### First query:
image

### Second query:
image


_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @ybadr99, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Populated database with data ✔️
- Created index on required columns ✔️
- Use of the explain analyze command ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**